alerting/src/test/kotlin/org/opensearch/alerting/MonitorDataSourcesIT.kt

```diff
@@ -209,6 +209,86 @@ class MonitorDataSourcesIT : AlertingSingleNodeTestCase() {
         assertEquals("Didn't match all 8 queries", 8, findings[0].docLevelQueries.size)
     }
 
+    fun `test execute monitor with nested json doc as source`() {
+        val docQuery1 = DocLevelQuery(query = "source.device.port:12345 OR source.device.hwd.id:12345", name = "3")
+
+        val docLevelInput = DocLevelMonitorInput(
+            "description", listOf(index), listOf(docQuery1)
+        )
+        val trigger = randomDocumentLevelTrigger(condition = ALWAYS_RUN)
+        val customFindingsIndex = "custom_findings_index"
+        val customFindingsIndexPattern = "custom_findings_index-1"
+        val customQueryIndex = "custom_alerts_index"
+        var monitor = randomDocumentLevelMonitor(
+            inputs = listOf(docLevelInput),
+            triggers = listOf(trigger),
+            dataSources = DataSources(
+                queryIndex = customQueryIndex,
+                findingsIndex = customFindingsIndex,
+                findingsIndexPattern = customFindingsIndexPattern
+            )
+        )
+        val monitorResponse = createMonitor(monitor)
+
+        val mappings = """{
+            "properties": {
+                "source.device.port": { "type": "long" },
+                "source.device.hwd.id": { "type": "long" }
+            }
+        }"""
+        client().admin().indices().putMapping(PutMappingRequest(index).source(mappings, XContentType.JSON)).get()
+        // testing both, nested and flatten documents
+        val testDocuments = mutableListOf<String>()
+        testDocuments += """{
+            "source" : { "device": {"port" : 12345 } }
+        }"""
+        testDocuments += """{
+            "source.device.port" : 12345
+        }"""
+        testDocuments += """{
+            "source" : { "device": {"hwd": { "id": 12345 } } }
+        }"""
+        testDocuments += """{
+            "source.device.hwd.id" : 12345
```

Comment:
what happens if we ingest "12345" i.e. a value in string literal which is parsable as a numeric

Reply:
added test like this